pkg/config/analysis/analyzer_test.go

```diff
@@ -79,63 +73,12 @@ func TestCombinedAnalyzer(t *testing.T) {
 	avalableSchemas.Add(&testSchemaImpl{col2.Name()})
 	avalableSchemas.Add(&testSchemaImpl{col3.Name()})
 
-	removed := a.RemoveSkipped(
-		avalableSchemas.Build(),
-		collection.Names{col3.Name()},
-		transformer.Providers{xform})
-
-	g.Expect(removed).To(ConsistOf(a3.Metadata().Name, a4.Metadata().Name))
-	g.Expect(a.Metadata().Inputs).To(ConsistOf(col1.Name(), col2.Name()))
-
 	a.Analyze(&context{})
 
 	g.Expect(a1.ran).To(BeTrue())
 	g.Expect(a2.ran).To(BeTrue())
-	g.Expect(a3.ran).To(BeFalse())
-	g.Expect(a4.ran).To(BeFalse())
-}
-
-// ProcessorOptions are options that are passed to event.Processors during startup.
-type ProcessorOptions struct {
-	MeshConfig   *v1alpha1.MeshConfig
-	DomainSuffix string
-}
-
-func TestGetDisabledOutputs(t *testing.T) {
-	g := NewWithT(t)
-
-	in1 := newSchema("in1")
-	in2 := newSchema("in2")
-	in3 := newSchema("in3")
-	in4 := newSchema("in4")
-	in5 := newSchema("in5")
-	out1 := newSchema("out1")
-	out2 := newSchema("out2")
-	out3 := newSchema("out3")
-	out4 := newSchema("out4")
-
-	blankFn := func(_ util.ProcessorOptions) event.Transformer {
-		return event.NewFnTransform(collection.SchemasFor(), collection.SchemasFor(), func() {}, func() {}, func(e event.Event, handler event.Handler) {})
-	}
-
-	xformProviders := transformer.Providers{
-		transformer.NewProvider(collection.SchemasFor(in1), collection.SchemasFor(out1, out2), blankFn),
-		transformer.NewProvider(collection.SchemasFor(in2), collection.SchemasFor(out3), blankFn),
-		transformer.NewProvider(collection.SchemasFor(in3), collection.SchemasFor(out3), blankFn),
-		transformer.NewProvider(collection.SchemasFor(in4, in5), collection.SchemasFor(out4), blankFn),
-	}
-
-	expectCollections(g, getDisabledOutputs(collection.Names{in1.Name()}, xformProviders), collection.Names{out1.Name(), out2.Name()})
-	expectCollections(g, getDisabledOutputs(collection.Names{in2.Name()}, xformProviders), collection.Names{})
-	expectCollections(g, getDisabledOutputs(collection.Names{in2.Name(), in3.Name()}, xformProviders), collection.Names{out3.Name()})
-	expectCollections(g, getDisabledOutputs(collection.Names{in4.Name()}, xformProviders), collection.Names{out4.Name()})
-}
-
-func expectCollections(g *GomegaWithT, actualSet map[collection.Name]struct{}, expectedCols collection.Names) {
-	g.Expect(actualSet).To(HaveLen(len(expectedCols)))
-	for _, col := range expectedCols {
-		g.Expect(actualSet).To(HaveKey(col))
-	}
+	g.Expect(a3.ran).To(BeTrue())
+	g.Expect(a4.ran).To(BeTrue())
```

Comment:
This is concerning.  The point of this test is to have analyzers requiring collections 1-4, but only provide collections 1-3, meaning analyzer 4 should not be run.  Inverting the success criteria of a test looks like perhaps a bug?